src/runtime/refiner.ts

```diff
@@ -13,6 +13,12 @@ import {Dictionary} from './hot.js';
 import {Schema} from './schema.js';
 import {Entity} from './entity.js';
 
+enum Primitive {
```

Comment:
Do we lose anything from this if we use `type Primitive = 'boolean' | 'number' | 'text';`?

There seems to be a lot of Primitive.BOOLEAN style code below. Maybe we could at least define some (non-exported) aliases to these so we don't have so much `Primitive.`?

Reply:
> Do we lose anything from this if we use type Primitive = 'boolean' | 'number' | 'text';?

There are instances where evalType can only be a subset of these as per my understanding, for instance, unary-expression can only be evaluated to a number or boolean but not as a text, even when we do provide Text support ('not' TEXT or '-' TEXT don't have a lot of meaning to them?). So I'm not sure if we can use type Primitive as a whole without ending up defining subset cases (which happens to be a fair few of them) explicitly as 'boolean' | 'number'.

>  There seems to be a lot of Primitive.BOOLEAN style code below. Maybe we could define some (non-exported) aliases to these so we don't have so much Primitive.?

What would an alias look like? I imagine we would still need some information in the alias to tell if it's a primitive or not, otherwise it doesn't say if we're referring to a boolean-ast-node, boolean-expression-node or boolean-primitive at face-value. We could use something like pBOOLEAN = Primitive.Boolean perhaps?

I am fairly sure I am not grasping your suggestion correctly. Would appreciate some elaboration.